assets/src/hooks/useDetour.ts

```diff
@@ -27,15 +27,21 @@ export const useDetour = (input: UseDetourInput) => {
 
   // Record snapshots when changed
   useEffect(() => {
-    const snapshotSubscription = actorRef.subscribe(() => {
-      const serializedSnapshot = JSON.stringify(actorRef.getPersistedSnapshot())
+    const snapshotSubscription = actorRef.subscribe((snap) => {
+      const persistedSnapshot = actorRef.getPersistedSnapshot()
+      const serializedSnapshot = JSON.stringify(persistedSnapshot)
       localStorage.setItem("snapshot", serializedSnapshot)
+      putDetourUpdate(persistedSnapshot).then((uuid: string | null) => {
+        if (uuid && snap.matches({ UUID: "Unset" })) {
+          send({ type: "detour.set.uuid", uuid })
+        }
```

Comment:
I don't think you need this. _maybe_ the UUID,
```suggestion
        if (uuid) {
          send({ type: "detour.set.uuid", uuid })
        }
```

but if we should _always_ get a UUID, then it should always be possible to send it right?
```suggestion
    send({ type: "detour.set.uuid", uuid })
```

[(I just realized) Maybe we should change out the type, because I think `null` is representing an error state here?](https://github.com/mbta/skate/pull/2729/files#r1711449321)

```suggestion
        if (isOk(uuid)) {
          send({ type: "detour.set.uuid", uuid: uuid.ok })
        }
```

Reply:
I think this leads to a loop? The snapshotSubscription will trigger whenever anything with the snapshot changes, then will try to update the detour, and the detour update will try to update the snapshot, triggering the subscription again. My goal with the `snap.matches` was to only permit the initial set ID change

Reply:
We discussed on a slack huddle and decided yeah, the `matches` is annoyingly needed. Refactored anyway because of different issues